diff --git a/arch/arm/boot/dts/qcom/sdm660-thermal.dtsi b/arch/arm/boot/dts/qcom/sdm660-thermal.dtsi
index e0ceaa108847c..922bccaeec09d 100644
--- a/arch/arm/boot/dts/qcom/sdm660-thermal.dtsi
+++ b/arch/arm/boot/dts/qcom/sdm660-thermal.dtsi
@@ -21,44 +21,44 @@
 		qcom,start-delay = <60>;
 		zone@0 {
 			reg = <0>;
-			qcom,silver-khz = <1612800>;
-			qcom,gold-khz = <1843200>;
-			qcom,trip-deg = <53>;
+			qcom,silver-khz = <1843200>;
+			qcom,gold-khz = <2208000>;
+			qcom,trip-deg = <55>;
 		};
 
 		zone@1 {
 			reg = <1>;
-			qcom,silver-khz = <1612800>;
-			qcom,gold-khz = <1843200>;
-			qcom,trip-deg = <58>;
+			qcom,silver-khz = <1843200>;
+			qcom,gold-khz = <2208000>;
+			qcom,trip-deg = <60>;
 		};
 
 		zone@2 {
 			reg = <2>;
-			qcom,silver-khz = <1536000>;
+			qcom,silver-khz = <1843200>;
 			qcom,gold-khz = <1747200>;
-			qcom,trip-deg = <62>;
+			qcom,trip-deg = <64>;
 		};
 
 		zone@3 {
 			reg = <3>;
 			qcom,silver-khz = <1401600>;
-			qcom,gold-khz = <1747200>;
-			qcom,trip-deg = <65>;
+			qcom,gold-khz = <1401600>;
+			qcom,trip-deg = <67>;
 		};
 
 		zone@4 {
 			reg = <4>;
 			qcom,silver-khz = <1113600>;
-			qcom,gold-khz = <1401600>;
-			qcom,trip-deg = <68>;
+			qcom,gold-khz = <1113600>;
+			qcom,trip-deg = <69>;
 		};
 
 		zone@5 {
 			reg = <5>;
 			qcom,silver-khz = <902400>;
 			qcom,gold-khz = <1113600>;
-			qcom,trip-deg = <70>;
+			qcom,trip-deg = <71>;
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/qcom/sdm660.dtsi b/arch/arm/boot/dts/qcom/sdm660.dtsi
index 3e115af7bc4d7..d6c18220eae3d 100644
--- a/arch/arm/boot/dts/qcom/sdm660.dtsi
+++ b/arch/arm/boot/dts/qcom/sdm660.dtsi
@@ -50,7 +50,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x0>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile0>;
 			qcom,lmh-dcvs = <&lmh_dcvs0>;
@@ -81,7 +81,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x1>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile0>;
 			qcom,lmh-dcvs = <&lmh_dcvs0>;
@@ -106,7 +106,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x2>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile0>;
 			qcom,lmh-dcvs = <&lmh_dcvs0>;
@@ -131,7 +131,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x3>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile0>;
 			qcom,lmh-dcvs = <&lmh_dcvs0>;
@@ -156,7 +156,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x100>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile1>;
 			qcom,lmh-dcvs = <&lmh_dcvs1>;
@@ -185,7 +185,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x101>;
-			clock-frequency = <1804800000>;
+			clock-frequency = <2208000000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile2>;
 			qcom,lmh-dcvs = <&lmh_dcvs1>;
@@ -210,7 +210,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x102>;
-			clock-frequency = <1804800000>;
+			clock-frequency = <2208000000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile3>;
 			qcom,lmh-dcvs = <&lmh_dcvs1>;
@@ -235,7 +235,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x103>;
-			clock-frequency = <1804800000>;
+			clock-frequency = <2208000000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile4>;
 			qcom,lmh-dcvs = <&lmh_dcvs1>;
@@ -300,12 +300,14 @@
 
 		CPU_COST_0: core-cost0 {
 			busy-cost-data = <
-				 633600 4
-				 902400 9
-				1113600 14
-				1401600 23
-				1536000 32
-				1612800 40
+				 633600   23
+				 902400  139
+				1113600  259
+				1401600  482
+				1536000  690
+				1612800  725
+				1747200  870
+				1843200 1102
 			>;
 			idle-cost-data = <
 				20 16 12 8
@@ -313,10 +315,14 @@
 		};
 		CPU_COST_1: core-cost1 {
 			busy-cost-data = <
-				1113600 111
-				1401600 140
-				1747200 174
-				1804800 180
+				1113600  749
+				1401600 1443
+				1747200 2576
+				1804800 2661
+				1958400 4153
+				2150400 5941
+				2208000 6688
+				2457600 7444
 			>;
 			idle-cost-data = <
 				100 80 60 40
@@ -324,12 +330,14 @@
 		};
 		CLUSTER_COST_0: cluster-cost0 {
 			busy-cost-data = <
-				 633600 3
-				 902400 8
-				1113600 13
-				1401600 23
-				1536000 32
-				1612800 39
+				 633600  8
+				 902400 14
+				1113600 20
+				1401600 29
+				1536000 35
+				1612800 37
+				1747200 49
+				1843200 62
 			>;
 			idle-cost-data = <
 				4 3 2 1
@@ -337,10 +345,14 @@
 		};
 		CLUSTER_COST_1: cluster-cost1 {
 			busy-cost-data = <
-				1113600 34
-				1401600 63
-				1747200 111
-				1804800 178
+				1113600 135
+				1401600 193
+				1747200 272
+				1804800 281
+				1958400 363
+				2150400 519
+				2208000 585
+				2457600 651
 			>;
 			idle-cost-data = <
 				4 3 2 1
@@ -1295,7 +1307,7 @@
 		qcom,core-dev-table =
 			< 902400 762 >,
 			< 1401600 2086 >,
-			< 1881600 3879 >;
+			< 1843200 3879 >;
 	};
 
 	devfreq_memlat_4: qcom,arm-memlat-mon-4 {
@@ -1306,7 +1318,7 @@
 			< 1113600 762 >,
 			< 1401600 3879 >,
 			< 2150400 5931 >,
-			< 2457600 6881 >;
+			< 2208000 6881 >;
 	};
 
 	devfreq_cpufreq: devfreq-cpufreq {
@@ -1368,7 +1380,8 @@
 			<  1113600000 0x0404003a 0x052e002e 0x2 4 >,
 			<  1401600000 0x04040049 0x073a003a 0x2 5 >,
 			<  1536000000 0x04040050 0x08400040 0x2 6 >,
-			<  1612800000 0x04040054 0x09430043 0x2 7 >;
+			<  1612800000 0x04040054 0x09430043 0x2 7 >,
+			<  1843200000 0x04040060 0x094c004c 0x3 8 >;
 
 		qcom,pwrcl-speedbin4-v0 =
 			<   300000000 0x0004000f 0x01200020 0x1 1 >,
@@ -1403,14 +1416,16 @@
 			<  1113600000 0x0404003a 0x052e002e 0x1 2 >,
 			<  1401600000 0x04040049 0x073a003a 0x2 3 >,
 			<  1747200000 0x0404005b 0x09480048 0x2 4 >,
-			<  1804800000 0x0404005e 0x094b004b 0x2 5 >;
+			<  1804800000 0x0404005e 0x094b004b 0x2 5 >,
+			<  2208000000 0x04040073 0x0b5c005c 0x3 6 >;
 
 		qcom,perfcl-speedbin4-v0 =
 			<   300000000 0x0004000f 0x01200020 0x1 1 >,
 			<  1113600000 0x0404003a 0x052e002e 0x1 2 >,
 			<  1401600000 0x04040049 0x073a003a 0x2 3 >,
 			<  1747200000 0x0404005b 0x09480048 0x2 4 >,
-			<  1958400000 0x04040066 0x0a510051 0x2 5 >;
+			<  1958400000 0x04040066 0x0a510051 0x2 5 >,
+			<  2208000000 0x04040073 0x0b5c005c 0x3 6 >;
 
 		qcom,up-timer = <1000 1000>;
 		qcom,down-timer = <1000 1000>;

